File: content/skills/agility/pyramid/src/main/java/org/jesse/game/content/skills/agility/pyramid/Doorway.java
package org.jesse.game.content.skills.agility.pyramid;

import org.jesse.game.content.skills.agility.AgilityCourseObstacle;
import org.jesse.game.world.entity.Location;
import org.jesse.game.world.entity.player.Player;
import org.jesse.game.world.object.WorldObject;
import org.jesse.plugins.dialogue.PlainChat;

public final class Doorway extends AgilityCourseObstacle {

    public Doorway() {
        super(AgilityPyramid.class, 7);
    }

    @Override
    public int getDuration(boolean success, WorldObject object) {
        return 0;
    }

    @Override
    public void startSuccess(Player player, WorldObject object) {
        player.setLocation(new Location(3364, 2830, 0));
        player.getDialogueManager().start(new PlainChat(player, "You climb down the steep passage. It leads to the base of the<br>pyramid"));
        player.getVarManager().sendBit(AgilityPyramid.HIDE_PYRAMID_VARBIT, false);
    }

    @Override
    public double getSuccessXp(WorldObject object) {
        return 0;
    }

    @Override
    public int getLevel(WorldObject object) {
        return 30;
    }

    @Override
    public int[] getObjectIds() {
        return new int[] { 10855, 10856 };
    }

}

File: content/skills/agility/pyramid/src/main/kotlin/org/jesse/game/content/skills/agility/pyramid/Doorway.kt
package org.jesse.game.content.skills.agility.pyramid

import org.jesse.game.content.skills.agility.AgilityCourseObstacle
import org.jesse.game.world.entity.Location
import org.jesse.game.world.entity.player.Player
import org.jesse.game.world.`object`.WorldObject
import org.jesse.plugins.dialogue.PlainChat

class Doorway : AgilityCourseObstacle(AgilityPyramid::class.java, 7) {
    override fun getDuration(success: Boolean, `object`: WorldObject?): Int {
        return 0
    }

    override fun startSuccess(player: Player, `object`: WorldObject?) {
        player.setLocation(Location(3364, 2830, 0))
        player.getDialogueManager()
            .start(PlainChat(player, "You climb down the steep passage. It leads to the base of the<br>pyramid"))
        player.getVarManager().sendBit(AgilityPyramid.Companion.HIDE_PYRAMID_VARBIT, false)
    }

    override fun getSuccessXp(`object`: WorldObject?): Double {
        return 0.0
    }

    override fun getLevel(`object`: WorldObject?): Int {
        return 30
    }

    override fun getObjectIds(): IntArray = intArrayOf(10855, 10856)
}
